src/paulscode/android/mupen64plusae/task/CacheRomInfoTask.java

```diff
@@ -103,6 +104,7 @@ public CacheRomInfoTask( Activity activity, File searchPath, String databasePath
     private final boolean mClearGallery;
     private final CacheRomInfoListener mListener;
     private final ProgressDialog mProgress;
+    private final Activity mActivity;
```

Comment:
Rather than caching the activity and doing it this way, I'd prefer to just add another public method to ProgressDialog.java:

```
    public void setMessage( final int resid )
    {
        mActivity.runOnUiThread( new Runnable()
        {
            @Override
            public void run()
            {
                mTextMessage.setText( resid );
            }
        } );
    }
    ```